### ISSUE TITLE
Term::Readline displays wrong prompt

### ISSUE THREAD

**Issue by p5pRT** [Perl/perl5#6076]
Migrated from [rt.perl.org#18421](https://rt-archive.perl.org/perl5/Ticket/Display.html?id=18421) (status was 'resolved')

Searchable as RT18421$

**Comment by p5pRT** [comment]
### From ewusaer@sd.us.am.ericsson.se

#### Created by ewusaer@sd.us.am.ericsson.se



When I use Term&#8203;::Readline with ansi colors\, the prompt displays the
last two characters twice\. i\.e\. instead of

 Enter your arithmetic expression&#8203;: 

I get&#8203;:

 Enter your arithmetic expression&#8203;: : 



My code is as follows&#8203;: 

\#\!/pkg/perl5plus/bin/perl 

use Term&#8203;::ReadLine;  



\# Colors\!\!
my $red     = "\\e\[31m";
my $green   = "\\e\[32m";
my $blue    = "\\e\[34m";
my $magenta = "\\e\[35m";
my $reset   = "\\e\[0m";



$term = new Term&#8203;::ReadLine 'Simple Perl calc';
$prompt = "$blue Enter your arithmetic expression$reset&#8203;: ";  
$OUT = $term\->OUT || STDOUT;
while \( defined \($\_ = $term\->readline\($prompt\)\) \) \{  
&nbsp;    $res = eval\($\_\)\, "\\n";
&nbsp;    warn $@&#8203; if $@&#8203;;    print $OUT $res\, "\\n" unless $@&#8203;;
&nbsp;    $term\->addhistory\($\_\) if /\\S/;  
\}

<details><summary>Perl Info</summary>

```


Site configuration information for perl 5.00404:

Configured by lsherwin at Thu Oct 22 09:06:12 PDT 1998.

Summary of my perl5 (5.0 patchlevel 4 subversion 4) configuration:
  Platform:
    osname=solaris, osvers=2.5.1, archname=sun4-solaris
    uname='sunos menehune 5.5.1 generic_103640-18 sun4u sparc sunw,ultra-1 '
    hint=recommended, useposix=true, d_sigaction=define
    bincompat3=y useperlio=undef d_sfio=undef
  Compiler:
    cc='gcc', optimize='-O', gccversion=2.7.2.1
    cppflags='-I/usr/local/include -I/pkg/gnu/include'
    ccflags ='-I/usr/local/include -I/pkg/gnu/include'
    stdchar='unsigned char', d_stdstdio=define, usevfork=false
    voidflags=15, castflags=0, d_casti32=define, d_castneg=define
    intsize=4, alignbytes=8, usemymalloc=y, prototype=define
  Linker and Libraries:
    ld='gcc', ldflags =' -L/usr/local/lib'
    libpth=/usr/local/lib /lib /usr/lib /usr/ccs/lib /pkg/gnu/lib
    libs=-lsocket -lnsl -ldl -lm -lc -lcrypt
    libc=/lib/libc.so, so=so
    useshrplib=false, libperl=libperl.a
  Dynamic Linking:
    dlsrc=dl_dlopen.xs, dlext=so, d_dlsymun=undef, ccdlflags=' '
    cccdlflags='-fpic', lddlflags='-G -L/usr/local/lib'

Locally applied patches:
	


@INC for perl 5.00404:
	/pkg/perl5plus/lib/sun4-solaris/5.00404
	/pkg/perl5plus/lib
	/pkg/perl5plus/lib/site_perl/sun4-solaris
	/pkg/perl5plus/lib/site_perl
	.


Environment for perl 5.00404:
    HOME=/usr2/ewusaer
    LANG=en_US
    LC_COLLATE=en_US
    LC_CTYPE=en_US
    LC_MESSAGES=C
    LC_MONETARY=en_US
    LC_NUMERIC=en_US
    LC_TIME=en_US
    LD_LIBRARY_PATH=/opt/mis/oracle/product/8.1.6/lib:/opt/mis/oracle/product/8.1.6/lib:/usr/lib:/usr/ucblib:/usr/openwin/lib:/usr/dt/lib
    LOGDIR (unset)
    PATH=/opt/mis/oracle/product/8.1.6/bin:/usr/atria/bin:/vobs/cth/com/project_space/bin:/opt/mis/oracle/product/8.1.6/bin:/usr/atria/bin:/vobs/cth/com/project_space/bin:/opt/SUNWsunpc/bin:/usr/atria/bin:/usr/bin:/usr/ccs/bin:/usr/ucb:/usr/afsws/bin:/usr/openwin/bin:/usr/dt/bin:/prj/bsc/bin:/usr/local/bin:.:/pkg/netscape/bin:/pkg/cbstools/bin:/pkg/ngoda/Drop7/bin:/prj/bsc/bin:/usr2/ewusaer/bin
    PERL_BADLANG (unset)
    SHELL=/bin/tcsh

```
</details>

**Comment by p5pRT** [comment]
### From @hvds

Salih Ergut \(via RT\) \<perlbug@&#8203;perl\.org> wrote&#8203;:
:When I use Term&#8203;::Readline with ansi colors\, the prompt displays the
:last two characters twice\. i\.e\. instead of
:
: Enter your arithmetic expression&#8203;: 
:
:I get&#8203;:
:
: Enter your arithmetic expression&#8203;: : 

This doesn't seem ever to have been answered\, for which my apologies\.

You are using a very old version of perl \(v5\.004\_04\)\. The problem you
report persists as far as v5\.005\_03\, but does not appear in v5\.6\.0
and later\. The current latest stable version of perl is v5\.8\.0\.
I don't know offhand which change fixed the problem \- it doesn't
appear to have been fixed by a direct patch to Term&#8203;::ReadLine \- so
I don't know whether there exists a fix that would allow you to
continue using v5\.004\_04\.

Hope this helps\,

Hugo van der Sanden

**Comment by p5pRT** [comment]
p5p@spam.wizbit.be - Status changed from 'open' to 'resolved'
